src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -163,6 +168,51 @@ public void Dispose()
             }
         }
 
+        public class DuplicateDocsHandler : IDisposable
+        {
+            public List<DocumentItem> DocumentsWithDuplicateCollection;
+            private readonly DocumentsOperationContext _context;
+            private IDisposable _disposable;
+            internal bool _throwOnCollectionMismatchError = false;
+
+            public DuplicateDocsHandler(DocumentDatabase database)
+            {
+                _disposable = database.DocumentsStorage.ContextPool.AllocateOperationContext(out _context);
+                DocumentsWithDuplicateCollection = new List<DocumentItem>();
+            }
+
+            internal void WriteDocumentWithDuplicateCollection(DocumentItem item)
+            {
+                var doc = new Document
```

Comment:
create a method inside document `Clone(JsonOperationContext context)` and handle everything there. Less error prone in future.